### PR TITLE
Updated Android SDK, all dependencies, and did the gradle migarion, fixed unit tests and ktlint 

### PR DESCRIPTION
**Pull request by artakka**
Updated Android SDK from 28 to 33
Updated all dependencies to latest versions
Fixed deprecated unit tests
Fixed line error by updated ktlint

### CODE REVIEW

**Review by birkskyum — APPROVED** [review]
Thanks! It seem to run as expected on SDK 33 where I tested, and the breaking node ci shouldn't have been triggered by changes to these directories in the first place.

The minSdkVersion of 14 is probably a bit too low at this point where SDK 14 and 15 are obsolete. What is the lowest SDK version that can run this?

File: platform/android/MapboxGLAndroidSDK/local.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Tue Dec 20 16:17:15 PST 2022
+sdk.dir=/Users/artak/Library/Android/sdk
```
Comment (wipfli) [inline platform/android/MapboxGLAndroidSDK/local.properties:8]: Probably you can remove this file again

File: platform/android/MapboxGLAndroidSDKLint/local.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Tue Dec 20 16:17:45 PST 2022
+sdk.dir=/Users/artak/Library/Android/sdk
```
Comment (wipfli) [inline platform/android/MapboxGLAndroidSDKLint/local.properties:8]: Probably you can remove this file again

**Comment by wipfli** [comment]
Oh did you add an entry in the changelog?

**Comment by artakka** [comment]
@wipfli it is in https://github.com/maplibre/maplibre-gl-native/pull/630

**Pull request merged**